oci/pull.bzl

```diff
@@ -0,0 +1,99 @@
+"Pull image layers using Bazel downloader"
+
+_attrs = {
+    "image": attr.string(),
+    "tag": attr.string(),
+}
+
+def _download(rctx, tag, output, type = "manifests"):
+    if type != "blobs" and type != "manifests":
+        fail("type must be blobs or manifests")
+    if rctx.attr.image.startswith("gcr.io/"):
+        registry_url = "https://gcr.io/v2/{image}/{type}/{tag}".format(
+            image = rctx.attr.image[len("gcr.io/"):],
+            type = type,
+            tag = tag,
+        )
+    else:
+        fail("Unrecognized registry. We only understand gcr.io currently. File an issue on rules_oci.")
```

Comment:
all v2-compatible registries follow the same URL semantics.  this could easily be a string attribute that defaults to gcr.io